### DISCUSSION TITLE
Is it possible to trigger an AnimationBehavior from a ViewModel

### DISCUSSION THREAD

**Discussion by bijington** [xamarin/XamarinCommunityToolkit#1156] [Ideas]
I would like to perform a `ShakeAnimation` on a `Button` if I cannot execute the `Command` in the ViewModel rather than just disabling the button. Is it possible to trigger the animation from the ViewModel side? As far as I can tell `AnimationBehavior` currently relies on events which I don't think helps me in this case.

**Comment by Cfun1** [comment]
I didn't look at ShakeAnimation implementation yet but according to your statement, just guessing (i might be wrong) that [EventToCommandBehavior](https://docs.microsoft.com/en-us/xamarin/community-toolkit/behaviors/eventtocommandbehavior) may be useful in this case.

**Reply by bijington** [reply]
Thanks for the quick response!

I am not sure it does help me unless my understanding is completely wrong. As I understand it `EventToCommandBehavior` still deals with something happening in the View and then it being able to perform something within the ViewModel. Is that correct?

Perhaps what I want to achieve or at least how I thought to achieve it was wrong... essentially I have a number of `Entry` controls that have some level of validation performed on them. I then have a create button which when press I would like to either shake all the invalid `Entry` controls or at least just the create button. I thought this would be triggered from the ViewModel.

**Comment by bijington** [comment]
This terrible drawing shows what I am thinking:

![image](https://user-images.githubusercontent.com/17139988/113418666-d5ccc180-93bd-11eb-8644-ec86ef9321e7.jpeg)

**Reply by AndreiMisiukevich** [reply]
No, now it's not possible. But it can be quite easily added, I am sure

**Reply by AndreiMisiukevich** [reply]
if you wish, you can open an issue with feature-request.)

**Reply by bijington** [reply]
@AndreiMisiukevich great thanks for the confirmation. I will raise it as a feature request and I'll be happy to provide the effort in implementing it 🙂

**Reply by AndreiMisiukevich** [reply]
> @AndreiMisiukevich great thanks for the confirmation. I will raise it as a feature request and I'll be happy to provide the effort in implementing it 🙂

awesome)

It should be a command with OneWayToSource binding mode (You can check how I implemented it in Expander - ForceUpdateSizeCommand)

**Reply by bijington** [reply]
Great thanks. I will follow that
